### PR TITLE
feat: Create an ErrorView that can be used to display errors

### PR DESCRIPTION
**Pull request by mofojed**
- Added to the StyleGuide as well
- Displays errors in a textarea, with a copy button and an expand button

### CODE REVIEW

File: packages/components/src/ErrorView.scss [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    width: 100%;
+
+    .error-view-header-text {
+      display: flex;
+      flex-direction: row;
+      align-items: center;
+      padding-left: $spacer;
+      padding-right: $spacer;
+      font-weight: bold;
+      text-overflow: ellipsis;
```
Comment (dsmmcken) [inline packages/components/src/ErrorView.scss:33]: not needed

File: packages/components/src/ErrorView.scss [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+      .error-view-copy-button {
+        min-width: 3rem;
+      }
+
+      .error-view-expand-button {
+        svg {
+          margin-right: $spacer-1;
+        }
+      }
```
Comment (dsmmcken) [inline packages/components/src/ErrorView.scss:73]: remove, button has defautl margin on svg now

File: packages/components/src/ErrorView.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+ */
+function ErrorView({
+  message,
+  isExpanded: isExpandedProp = false,
+  type = 'Error',
+}: ErrorViewerProps): JSX.Element {
+  const [isExpandable, setIsExpandable] = useState(false);
+  const [isExpanded, setIsExpanded] = useState(false);
+  const viewRef = useRef<HTMLDivElement>(null);
+  const textareaRef = useRef<HTMLPreElement>(null);
```
Comment (dsmmcken) [inline packages/components/src/ErrorView.tsx:35]: rename, not a textarea

File: packages/components/src/ErrorView.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+  const handleResize = useCallback(() => {
+    if (isExpanded || textareaRef.current == null) {
```
Comment (dsmmcken) [inline packages/components/src/ErrorView.tsx:38]: should also check `isExpandedProp` ?

**Review by dsmmcken — CHANGES_REQUESTED** [review]
webkit styleguide snapshot looks wrong

File: packages/components/src/ErrorView.tsx (hunk 4 above)
Comment (mofojed) [reply packages/components/src/ErrorView.tsx:38]: Doesn't need to, `isExpandedProp` being true will never show the expand more/less button. Don't need to change this logic.

File: packages/components/src/ErrorView.scss [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      border-radius: 0 $border-radius 0 0;
+      flex-shrink: 0;
+
+      .btn-danger {
+        border-radius: 0;
+        color: $black;
+        opacity: 0.8;
+        padding: $spacer-1;
+        &:active {
+          color: $black;
```
Comment (dsmmcken) [inline packages/components/src/ErrorView.scss:60]:
```suggestion
        color: var(--dh-color-contrast-dark);
        opacity: 0.8;
        padding: $spacer-1;
        &:active {
          color: var(--dh-color-contrast-dark);
```

Got to watch those color variables for light theme.

**Review by dsmmcken — CHANGES_REQUESTED** [review]

File: packages/components/src/ErrorView.tsx (hunk 4 above)
Comment (dsmmcken) [reply packages/components/src/ErrorView.tsx]: Yeah but aren't you doing an unnecessary work still? Checking height can trigger a forced reflow.

File: packages/components/src/ErrorView.scss (hunk 5 above)
Comment (mattrunyon) [reply packages/components/src/ErrorView.scss:60]: We could try adding a stylelint rule to ban sass variables basically. I think we could use [this rule](https://github.com/stylelint-scss/stylelint-scss/blob/master/src/rules/dollar-variable-pattern/README.md) and set the regex to something like `/iWillNotUseSassVariables/` which would effectively ban sass variables I think

Comment (mofojed) [reply packages/components/src/ErrorView.scss:60]: Would need rules to allow for $spacer as well.

**Review by dsmmcken — APPROVED** [review]

**Pull request merged**